osm2streets/src/road.rs

```diff
@@ -109,7 +119,37 @@ impl Road {
             crossing_nodes: Vec::new(),
 
             lane_specs_ltr,
-        }
+        };
+
+        result.update_center_line(config.driving_side); // TODO delay this until trim_start and trim_end are calculated
+        result
+    }
+
+    /// Calculates and sets the center_line from reference_line, reference_line_placement
+    /// (and TODO trim_start, trim_end).
+    pub fn update_center_line(&mut self, driving_side: DrivingSide) {
+        let ref_position = match self.reference_line_placement {
+            Placement::Consistent(p) => p,
+            Placement::Varying(p, _) => {
+                warn!("varying placement not yet supported, using placement:start");
+                p
+            }
+            Placement::Transition => {
```

Comment:
Finally reading the Wiki...

> Instead of the value transition, one can use the [suffixes :start and :end](https://wiki.openstreetmap.org/wiki/Proposed_features/Suffix_start_and_end) to provide information about the placement at the start resp. end of the OSM.way. This is usually not necessary if such information can be retrieved from the previous resp. following OSM way.

To later support this correctly, we have to look at the previous and next OSM ways (and decide which one is "the same road" -- probably just by name?) and interpret explicit or implicit placement tags there?

Reply:
We look at the traffic flow lane by lane to determine the lane connections. With `Connection`s and `Forks` you can examine movements, do lane math and divvy up the lanes easily. Make some assumptions if its not trivial, as the mapper can tag the lane connectivity explicitly too.

I think we are supposed to tag `placement=transition` on the nonsense segments of motorway exits (the ones that go way outside the lanes that they represent, and so couldn't be tagged with any sensible `placement:start` anyway). We probably just assume them to be at the start of the applicable roads automatically, because we need to do it all over the place anyway.

When both Roads have their placement tagged (as non-transition), that actually tells us the lane connectivity at that node. Try mapping a segment where a right turn lane appears for a short distance and the ways stay straight with `right_of:1` for the 2 lane and 3 lane segments.